charts/kserve-resources/templates/configmap.yaml

```diff
@@ -199,9 +198,15 @@ data:
            # NOTE: This configuration only applicable for serverless deployment with Istio configured as network layer.
            "ingressGateway" : "knative-serving/knative-ingress-gateway",
 
-           # ingressService specifies the hostname of the ingress service.
+           # knativeLocalGatewayService specifies the hostname of the Knative's local gateway serivice.
+           # The default KServe configurations are re-using the Istio local gateways for Knative. In this case, this
+           # knativeLocalGatewayService field can be left unset. When unset, the value of "localGatewayService" will be used.
+           # However, sometimes it may be better to have local gateways specifically for KServe (e.g. when enabling strict mTLS in Istio).
+           # Under such setups where KServe is needed to have its own local gateways, the values of the "localGateway" and
+           # "localGatewayService" should point to the KServe local gateways. Then, this knativeLocalGatewayService field
+           # should point to the Knative's local gateway service.
```

Comment:
why need to point to knative's local gateway here even if there is a dedicated gateway for KServe?
I saw KNative support [Install Serving with YAML](https://knative.dev/docs/install/yaml-install/serving/install-serving-with-yaml/) and seems like it can support multiple entries? if we updated the dedicated KServe localgateway here as well, do we still need to point to knative local gateway?

Reply:
It is related to the top-level virtual service that KServe creates. There is some explanation here: https://kserve.github.io/website/latest/admin/serverless/servicemesh/#disable-top-level-virtual-service

Reply:
Can you explain a bit more how this works when top level virtual service is enabled by using an additional gateway? I am a bit concerned that this adds additional configuration complexity to kserve admin.

Reply:
If I understand correctly, the additional local gateway is needed for handling the non service mesh traffic. But I am not sure if this is a real use case as if the cluster is turned on for service mesh then mutual tls should not in cluster. You can use ingress gateway to handle the non service mesh traffic so we eliminate the need of the additional local gateway.

Reply:
> But I am not sure if this is a real use case as if the cluster is turned on for service mesh then mutual tls should work in cluster.

@yuzisun  Hopefully, it is more clear after our offline call.

For the record, and in short, when locking the mesh with mTLS using both _PeerAuthentication_ and also _DestinationRule_ traffic becomes blocked on the ingress-gateway when using the top-level virtual service. This is because the top-level virtual service is configuring a loop on the ingress gateway. The listening port of the ingress gateway is using plain-text. In the default configurations, this loopback connection of the top-level virtualservice is possible, because the connection happens as plain-text. However, when locking the mesh with a DestinationRule with mTLS, this loopback connection no longer succeeds because it is tried with mTLS while the gateway is still listening with plain-text.

By allowing KServe to have its own Gateway, it becomes possible to open a new port in the gateway that would listen with mTLS rather than plain-text. Thus, it becomes feasible to configure the top-level virtualservice to connect to the mTLS port.